openassessment/assessment/api/staff.py

```diff
@@ -0,0 +1,292 @@
+            information from the scores, an error is raised.
+    """
+    try:
+        assessments = list(
+            Assessment.objects.filter(
+                score_type=STAFF_TYPE, submission_uuid=submission_uuid
+            )[:1]
+        )
+        scores = Assessment.scores_by_criterion(assessments)
+        return Assessment.get_median_score_dict(scores)
+    except DatabaseError:
+        error_message = u"Error getting staff assessment scores for {}".format(submission_uuid)
+        logger.exception(error_message)
+        raise StaffAssessmentInternalError(error_message)
+
+
+def create_assessment(
+    submission_uuid,
+    scorer_id,
+    options_selected,
+    criterion_feedback,
+    overall_feedback,
+    rubric_dict,
+    scored_at=None
+):
+    """Creates an assessment on the given submission.
+
+    Assessments are created based on feedback associated with a particular
+    rubric.
+
+    Assumes that the user creating the assessment has the permissions to do so.
+
+    Args:
+        scorer_id (str): The user ID for the user giving this assessment. This
+            is required to create an assessment on a submission.
+        options_selected (dict): Dictionary mapping criterion names to the
+            option names the user selected for that criterion.
+        criterion_feedback (dict): Dictionary mapping criterion names to the
+            free-form text feedback the user gave for the criterion.
+            Since criterion feedback is optional, some criteria may not appear
+            in the dictionary.
+        overall_feedback (unicode): Free-form text feedback on the submission overall.
+
+    Keyword Args:
+        scored_at (datetime): Optional argument to override the time in which
+            the assessment took place. If not specified, scored_at is set to
+            now.
+
+    Returns:
+        dict: the Assessment model, serialized as a dict.
+
+    Raises:
+        StaffAssessmentRequestError: Raised when the submission_id is invalid, or
+            the assessment_dict does not contain the required values to create
+            an assessment.
+        StaffAssessmentInternalError: Raised when there is an internal error
+            while creating a new assessment.
+
+    Examples:
+        >>> options_selected = {"clarity": "Very clear", "precision": "Somewhat precise"}
+        >>> criterion_feedback = {"clarity": "I thought this essay was very clear."}
+        >>> feedback = "Your submission was thrilling."
+        >>> create_assessment("Tim", options_selected, criterion_feedback, feedback, rubric_dict)
+    """
+    try:
+        assessment = _complete_assessment(
+            submission_uuid,
+            scorer_id,
+            options_selected,
+            criterion_feedback,
+            overall_feedback,
+            rubric_dict,
+            scored_at
+        )
+        return full_assessment_dict(assessment)
+
+    except InvalidRubric:
+        msg = u"Rubric definition was not valid"
```

Comment:
Nit: I'm not sure why this is stored in a variable, but since it is I'd prefer it be called ``error_message`` like it is below.